dsgrid/dataset/dataset_schema_handler_base.py

```diff
@@ -260,29 +290,24 @@ def _remap_dimension_columns(self, df, pivoted_columns=None):
 
         return df
 
-    @track_timing(timer_stats_collector)
-    def _add_time_zone(self, load_data_df):
-        """Add time_zone col to load_data_df"""
-        geography_dim = self._config.get_dimension(DimensionType.GEOGRAPHY)
-        geo_records = geography_dim.get_records_dataframe()
-        geo_name = geography_dim.model.dimension_type.value
-        load_data_df = add_column_from_records(load_data_df, geo_records, geo_name, "time_zone")
-        return load_data_df
-
-    def get_time_zone_mapping(self):
-        """Get time_zone mapping to map to load_data_df
-        Returns
-        -------
-        pyspark.sql.DataFrame
-            a two-column df containing time_zone and a mapping key column
-        """
+    def _convert_time_before_project_mapping(self):
+        time_dim = self._config.get_dimension(DimensionType.TIME)
+        return (
+            time_dim.model.does_geography_require_time_zone()
+            and not self._config.model.use_project_geography_time_zone
+        )
 
     @track_timing(timer_stats_collector)
-    def _convert_time_dimension(self, load_data_df, time_zone_mapping):
+    def _convert_time_dimension(self, load_data_df, project_config):
         time_dim = self._config.get_dimension(DimensionType.TIME)
-        load_data_df = time_dim.convert_dataframe(
+        if time_dim.model.does_geography_require_time_zone():
```

Comment:
This is awkward for me because the checks exist both here and in _convert_time_before_project_mapping(). But I am following here, I think the better solution is to remove _convert_time_before_project_mapping() later.

Reply:
1. The checks are conflicting. `_convert_time_before_project_mapping is dependent on `config.model.use_project_geography_time_zone` but this function is not.
2. This function will always call `time_dim.convert_dataframe()`, and we don't want that called twice in one pass..